cluster.go

```diff
@@ -289,6 +312,74 @@ func parseSlots(slots RedisMessage, defaultAddr string) map[string]group {
 	return groups
 }
 
+// parseShards - map redis shards for each redis nodes/addresses
+// defaultAddr is needed in case the node does not know its own IP
+func parseShards(shards RedisMessage, defaultAddr string) map[string]group {
+	parseNodeEndpoint := func(msg map[string]RedisMessage) string {
+		var ip string
+		switch msg["ip"].string {
+		case "":
+			return defaultAddr
+		case "?":
+			return ""
+		default:
+			ip = msg["ip"].string
+		}
+
+		_, isPort := msg["port"]
+		_, isTlsPort := msg["tls-port"]
+		var port int64
+		switch {
+		case isPort && !isTlsPort:
+			port = msg["port"].integer
+		case !isPort && isTlsPort:
+			port = msg["tls-port"].integer
+		default:
+			// TODO which one to choose if both set?
```

Comment:
It should depend on whether the client enables TLS config or not.

Reply:
Okay I get it, One thing I am unsure of is whether normal ports should even be accepted if TLS has been set.

Reply:
Yes, I believe it should be accepted if there is no "tls-port" provided.